### PR TITLE
feat(vapp): Add resource `cloudavenue_vapp_vm_internal_disk`

### PR DESCRIPTION
**Pull request by eremid**
### Description of your changes

If you submit change in the provider code, please make sure to:

- [x] Write or modify examples in `examples/` directory
- [x] Write or modify acceptance tests
- [x] Run `make generate` to ensure the doc was updated properly

### How has this code been tested

```
TF_ACC=1 go test -v -count=1  -run TestAccVMInternalDiskResource  ./internal/tests
=== RUN   TestAccVMInternalDiskResource
--- PASS: TestAccVMInternalDiskResource (35.26s)
PASS
ok      github.com/orange-cloudavenue/terraform-provider-cloudavenue/internal/tests     35.368s
```

### CODE REVIEW

File: internal/provider/vm/internal_disk_resource.go [hunk 1]
```diff
@@ -0,0 +1,618 @@
...
+	VMName          types.String `tfsdk:"vm_name"`
+	BusType         types.String `tfsdk:"bus_type"`
+	SizeInMb        types.Int64  `tfsdk:"size_in_mb"`
+	BusNumber       types.Int64  `tfsdk:"bus_number"`
+	UnitNumber      types.Int64  `tfsdk:"unit_number"`
+	ThinProvisioned types.Bool   `tfsdk:"thin_provisioned"`
+	Iops            types.Int64  `tfsdk:"iops"`
+	StorageProfile  types.String `tfsdk:"storage_profile"`
+	AllowVMReboot   types.Bool   `tfsdk:"allow_vm_reboot"`
+}
```
Comment (azrod) [inline internal/provider/vm/internal_disk_resource.go:58]: This struct is necessary in `vapp`. 
we can move to common ?

File: internal/provider/vm/internal_disk_resource.go [hunk 2]
```diff
@@ -0,0 +1,618 @@
...
+					stringplanmodifier.UseStateForUnknown(),
+				},
+			},
+			"vapp_name": schema.StringAttribute{
+				Required: true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.RequiresReplace(),
+				},
+				MarkdownDescription: "The vApp this VM internal disk belongs to.",
+			},
```
Comment (azrod) [inline internal/provider/vm/internal_disk_resource.go:89]: - If `vapp_name` is required why the resource is not `cloudavenue_vapp_vm_internal_disk ` ? 
- It's possible to add `vapp_id` ?

File: internal/provider/vm/internal_disk_resource.go [hunk 3]
```diff
@@ -0,0 +1,618 @@
...
+					stringplanmodifier.RequiresReplace(),
+				},
+				MarkdownDescription: "VM in vApp in which internal disk is created.",
+			},
+			"bus_type": schema.StringAttribute{
+				Required: true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.RequiresReplace(),
+				},
+				MarkdownDescription: "The type of disk controller. Possible values: ide, parallel( LSI Logic Parallel SCSI), sas(LSI Logic SAS (SCSI)), paravirtual(Paravirtual (SCSI)), sata, nvme.",
```
Comment (azrod) [inline internal/provider/vm/internal_disk_resource.go:102]:
```suggestion
				MarkdownDescription: "The type of disk controller. Possible values: `ide`, `parallel`( LSI Logic Parallel SCSI), `sas`(LSI Logic SAS (SCSI)), `paravirtual`(Paravirtual (SCSI)), `sata`, `nvme`.",
```

File: internal/provider/vm/internal_disk_resource.go [hunk 4]
```diff
@@ -0,0 +1,618 @@
...
+}
+
+// Metadata returns the resource type name.
+func (r *internalDiskResource) Metadata(_ context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_vm_internal_disk"
+}
+
+// Schema defines the schema for the resource.
+func (r *internalDiskResource) Schema(ctx context.Context, _ resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = schema.Schema{
```
Comment (azrod) [inline internal/provider/vm/internal_disk_resource.go:67]: This schema is required in `cloudavenue_vapp_vm`. It's possible to move Schema in the `func` ?

File: internal/provider/vm/internal_disk_resource.go [hunk 5]
```diff
@@ -0,0 +1,618 @@
...
+					int64planmodifier.RequiresReplace(),
+				},
+				MarkdownDescription: "The number of the SCSI or IDE controller itself.",
+			},
+			"unit_number": schema.Int64Attribute{
+				Required: true,
+				PlanModifiers: []planmodifier.Int64{
+					int64planmodifier.RequiresReplace(),
+				},
+				MarkdownDescription: "The device number on the SCSI or IDE controller of the disk.",
```
Comment (azrod) [inline internal/provider/vm/internal_disk_resource.go:123]:
```suggestion
				MarkdownDescription: "The device number on the `SCSI` or `IDE` controller of the disk.",
```

File: internal/provider/vm/internal_disk_resource.go [hunk 6]
```diff
@@ -0,0 +1,618 @@
...
+			"size_in_mb": schema.Int64Attribute{
+				Required:            true,
+				MarkdownDescription: "The size of the disk in MB.",
+			},
+			"bus_number": schema.Int64Attribute{
+				Required: true,
+				PlanModifiers: []planmodifier.Int64{
+					int64planmodifier.RequiresReplace(),
+				},
+				MarkdownDescription: "The number of the SCSI or IDE controller itself.",
```
Comment (azrod) [inline internal/provider/vm/internal_disk_resource.go:116]:
```suggestion
				MarkdownDescription: "The number of the `SCSI` or `IDE` controller itself.",
```

File: internal/tests/vm_internal_disk_resource_test.go [hunk 7]
```diff
@@ -0,0 +1,103 @@
...
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/terraform"
+)
+
+//go:generate go run github.com/FrangipaneTeam/tf-doc-extractor@latest -filename $GOFILE -example-dir ../../examples -test
+const testAccVMInternalDiskResourceConfig = `
+resource "cloudavenue_vm_internal_disk" "example" {
+  vapp_name       = "vapp_test3"
+  vm_name         = "TestRomain"
```
Comment (azrod) [inline internal/tests/vm_internal_disk_resource_test.go:16]: Please change `TestRomain` to more generic

Comment (eremid) [reply internal/tests/vm_internal_disk_resource_test.go:16]: yes... waiting vapp/vm resources to do this more properly.

**Review by azrod — APPROVED** [review]

**Pull request merged**